app/models/user.rb

```diff
@@ -1,2 +1,6 @@
 class User < ApplicationRecord
+  def linking_tests(level)
+    Test.joins("INNER JOIN results On tests.id = results.test_id").
+        where("results.user_id = ?", self.id).and(Test.where(level: level))
```

Comment:
- "селф" тут явно не нужно указывать
- два условия можно писать внутри одного `where` через запятую

например здесь можно так решить эту строчку
```suggestion
        where(results: { user_id: id }, level: level)
```